src/client/http_client.rs

```diff
@@ -238,9 +238,9 @@ impl HTTPClient {
         request: ManagementRequest,
         deadline: Instant,
         ctrl_c: Arc<AtomicBool>,
-    ) -> Result<HttpResponse, ClientError> {
-        let rt = Runtime::new().unwrap();
-        rt.block_on(async {
+    ) -> Result<HttpStreamResponse, ClientError> {
+        let rt = Arc::new(Runtime::new().unwrap());
+        rt.clone().block_on(async {
```

Comment:
Why does this need to be cloned?

Reply:
For that matter why does it need to be Arc at all?

Reply:
Needs to be an Arc so that we can clone it. And needs to be cloned as we return it from the function as part of the HttpStreamResponse. Needs to be part of the HttpStreamResponse as if we try and read the stream from a different runtime we panic.